server/routerlicious/packages/services-utils/src/auth.ts

```diff
@@ -143,6 +147,95 @@ function getTokenFromRequest(request: Request): string {
 	return tokenMatch[1];
 }
 
+export async function verifyToken(
+	tenantId: string,
+	documentId: string,
+	token: string,
+	tenantManager: ITenantManager,
+	options: IVerifyTokenOptions,
+): Promise<void> {
+	if (options.requireDocumentId && !documentId) {
+		throw new NetworkError(403, "Missing documentId.");
+	}
+
+	let tokenLifetimeMs: number | undefined;
+	try {
+		const claims = validateTokenClaims(token, documentId, tenantId, options.requireDocumentId);
+		if (options.requireTokenExpiryCheck) {
+			if (!options.maxTokenLifetimeSec) {
+				const errorMessage = "Missing maxTokenLifetimeSec in options";
+				Lumberjack.error(errorMessage, getLumberBaseProperties(documentId, tenantId));
+				throw new Error(errorMessage);
+			}
+			tokenLifetimeMs = validateTokenClaimsExpiration(claims, options.maxTokenLifetimeSec);
+		}
+
+		// Check token cache first
+		if ((options.enableTokenCache || options.ensureSingleUseToken) && options.tokenCache) {
+			const tokenCacheKey = token;
+			const cachedToken = await options.tokenCache.get(tokenCacheKey).catch((error) => {
+				Lumberjack.error(
+					"Unable to retrieve cached JWT",
+					claims
+						? getLumberBaseProperties(claims.documentId, claims.tenantId)
```

Comment:
nit: consolidate these lumberproperties up to line 163-164? Instead of defining them per log?